Engine/Source/Timer.cpp

```diff
@@ -63,10 +82,49 @@ long Timer::ReadDelta() {
 	Uint32 elapsedTime = newTime - mLastReadTime;
 	mLastReadTime = newTime;
 	long convertedTime = static_cast<long>(elapsedTime * mSpeed);
+	mRealTime += elapsedTime;
 	mTotalTime += convertedTime;
 	return convertedTime;
 }
 
+long Timer::Stop() {
+	Uint64 finalTime = Read();
+
+	//We reset all variables
+	mLastReadTime = 0;
```

Comment:
Maybe we can encapsulate all this stop logic (which sets all the members from Timer to default/zero values, with a private function which can handle that. Is a small detail, but will make all the code cleaner.